src/portals/hope/features/claims/claim-details/ClaimNotes/use-claim-notes.tsx

```diff
@@ -0,0 +1,91 @@
+import { addSeconds } from 'date-fns'
+import {
+  ClaimNotesFragment,
+  useAddClaimNoteMutation,
+  useClaimNotesQuery,
+} from 'types/generated/graphql'
+import { useMe } from 'portals/hope/features/user/hooks/use-me'
+import { toast } from 'react-hot-toast'
+import formatDate from 'date-fns/format'
+import gql from 'graphql-tag'
+import { PushUserAction } from 'portals/hope/features/tracking/utils/tags'
+
+gql`
+  query ClaimNotes($claimId: ID!) {
+    claim(id: $claimId) {
+      id
+      notes {
+        text
+        date
+        handlerReference
+      }
+    }
+  }
+
+  mutation AddClaimNote($claimId: ID!, $note: ClaimNoteInput!) {
+    addClaimNote(id: $claimId, note: $note) {
+      id
+      ...ClaimNotes
+    }
+  }
+
+  fragment ClaimNotes on Claim {
+    notes {
+      text
+      date
+      handlerReference
+    }
+  }
+`
+
+interface UseClaimNotesResult {
+  notes: ClaimNotesFragment['notes']
+  createNote: (note: string) => void
+  creating: boolean
+}
+
+export const useClaimNotes = (claimId: string): UseClaimNotesResult => {
+  const [addClaimNote, { loading }] = useAddClaimNoteMutation()
+  const { data } = useClaimNotesQuery({
+    variables: { claimId },
+  })
+
+  const { me } = useMe()
+
+  const createNote = async (note: string) => {
+    const today = formatDate(addSeconds(new Date(), 1), 'yyyy-MM-dd HH:mm:ss')
+
+    PushUserAction('claim', 'add', 'note', null)
+
+    await toast.promise(
+      addClaimNote({
+        variables: { claimId, note: { text: note } },
+        optimisticResponse: {
+          addClaimNote: {
+            __typename: 'Claim',
+            id: claimId,
+            notes: [
+              {
+                text: note,
+                handlerReference: me.email ?? '',
+                date: today,
+              },
+              ...notes,
+            ],
+          },
+        },
+      }),
+      {
+        loading: 'Creating note...',
+        success: 'Note successfully created!',
+        error: 'Could not create note',
```

Comment:
Nitty UX copywriting 🙊 
```suggestion
        loading: 'Adding note...',
        success: 'Note successfully added!',
        error: 'Could not add note',
```